Workflow failure: Operator IT on OpenShift

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

```
2024-02-05 15:55:47  [ERROR] org.keycloak.operator.testsuite.integration.WatchedSecretsTest.testSecretChangesArePropagated  Time elapsed: 76.867 s  <<< ERROR!
2024-02-05 15:55:47  org.awaitility.core.ConditionTimeoutException: Condition with org.keycloak.operator.testsuite.integration.WatchedSecretsTest was not fulfilled within 1 minutes.
2024-02-05 15:55:47  	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:164)
2024-02-05 15:55:47  	at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
2024-02-05 15:55:47  	at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
2024-02-05 15:55:47  	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:939)
2024-02-05 15:55:47  	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:908)
2024-02-05 15:55:47  	at org.keycloak.operator.testsuite.integration.WatchedSecretsTest.testSecretChangesArePropagated(WatchedSecretsTest.java:92)
2024-02-05 15:55:47  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2024-02-05 15:55:47  	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:1013)
2024-02-05 15:55:47  	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:827)
2024-02-05 15:55:47  	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2024-02-05 15:55:47  	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

For full log see https://keycloak-jenkins/view/OpenShift/job/rhbk-operator-test-pipeline/303/execution/node/69/log/

### Version

latest

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java
@@ -70,6 +70,7 @@ import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.FileWriter;
 import java.lang.reflect.Method;
+import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.time.Duration;
@@ -342,6 +343,17 @@ public class BaseOperatorTest implements QuarkusTestAfterEachCallback {
           Log.warnf("%s not ready log: %s", instance.getMetadata().getName(), log);
       } catch (KubernetesClientException e) {
           Log.warnf("No %s log: %s", instance.getMetadata().getName(), e.getMessage());
+          if (instance instanceof Pod) {
+              try {
+                  String previous = k8sclient.raw(String.format("/api/v1/namespaces/%s/pods/%s/log?previous=true", namespace, instance.getMetadata().getName()));
+                  Log.warnf("%s previous log: %s", instance.getMetadata().getName(), previous);
+              } catch (KubernetesClientException pe) {
+                  // not available
+                  if (pe.getCode() != HttpURLConnection.HTTP_BAD_REQUEST) {
+                      Log.infof("Could not obtain previous log for %s: %s", instance.getMetadata().getName(), e.getMessage());
+                  }
+              }
+          }
       }
   }
 

--- a/operator/src/test/resources/example-postgres.yaml
+++ b/operator/src/test/resources/example-postgres.yaml
@@ -45,7 +45,6 @@ metadata:
 spec:
   selector:
     app: postgresql-db
-  type: LoadBalancer
   ports:
   - port: 5432
     targetPort: 5432

